pysrc/pip_resolve.py

```diff
@@ -145,34 +145,35 @@ def satisfies_python_requirement(parsed_operator, py_version_str, sys=sys):
 
 def get_markers_text(requirement):
     if isinstance(requirement, pipfile.PipfileRequirement):
-        markers_text = requirement.markers
-    else:
-        markers_text = requirement.line
-    return markers_text
+        return requirement.markers
+    return requirement.line
 
 def matches_python_version(requirement):
     """Filter out requirements that should not be installed
     in this Python version.
     See: https://www.python.org/dev/peps/pep-0508/#environment-markers
     """
     markers_text = get_markers_text(requirement)
-    if not markers_text:
-        return True
-    if not 'python_version' in markers_text:
+    if not (markers_text and re.match(".*;.*python_version", markers_text)):
         return True
-    match = PYTHON_MARKER_REGEX.search(markers_text)
-    if not match:
-        return False
-    parsed_operator = match.group('operator')
-    parsed_python_version = match.group('python_version')
-
-    if not parsed_python_version or not parsed_operator:
-        return False
-
-    return satisfies_python_requirement(
-        parsed_operator,
-        parsed_python_version
-    )
+
+    cond_text = markers_text.split(";", 1)[1]
+
+    # Gloss over the 'and' case and return true on the first matching python version
+
+    for sub_exp in re.split("\s*(?:and|or)\s*", cond_text):
```

Comment:
We basically split up the expressions by `or` and `and` and return positive on the first matching python version.